cmd/silkworm_api/snapshot_idx.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"github.com/ledgerwatch/erigon-lib/common/background"
+	"github.com/ledgerwatch/erigon-lib/common/datadir"
+	"github.com/ledgerwatch/erigon-lib/downloader/snaptype"
+	"github.com/ledgerwatch/erigon-lib/kv/mdbx"
+	"github.com/ledgerwatch/erigon/cmd/hack/tool/fromdb"
+	"github.com/ledgerwatch/erigon/turbo/debug"
+	"github.com/ledgerwatch/erigon/turbo/snapshotsync/freezeblocks"
+	"github.com/ledgerwatch/log/v3"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/sync/errgroup"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func main() {
+
+	app := &cli.App{
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "datadir",
+				Value: "./dev",
+				Usage: "node data directory",
+			},
+			&cli.StringFlag{
+				Name:  "snapshot_path",
+				Value: ".",
+				Usage: "pathname of the snapshot file",
+			},
+		},
+		Action: func(cCtx *cli.Context) error {
+			return buildIndex(cCtx, cCtx.String("datadir"), cCtx.String("snapshot_path"))
+		},
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		log.Crit(err.Error())
+	}
+}
+
+func FindIf(segments []snaptype.FileInfo, predicate func(snaptype.FileInfo) bool) (snaptype.FileInfo, bool) {
+	for _, segment := range segments {
+		if predicate(segment) {
+			return segment, true
+		}
+	}
+	return snaptype.FileInfo{}, false // Return zero value and false if not found
+}
+
+func buildIndex(cliCtx *cli.Context, dataDir string, snapshotPath string) error {
+	logger, err := debug.Setup(cliCtx, true /* rootLogger */)
+	if err != nil {
+		return err
+	}
+	logLevel := log.LvlInfo
+
+	ps := background.NewProgressSet()
+
+	workers := 4
+	g, ctx := errgroup.WithContext(cliCtx.Context)
+	g.SetLimit(workers)
```

Comment:
Now you can do: g.Go() and in the end
if err := g.Wait(); err != nil {
Return err
}

Reply:
@AskAlexSharov thank you for your suggestions!! I have just amended the code to use g.Go() and g.Wait()

Reply:
Amended command line to trigger multiple indexes creation